File: array/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        records = {}
        for i in range(len(nums)):
            s = target - nums[i]
            if s in records:
                return [records[s], i]
            records[nums[i]] = i

        return []

File: hash-table/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i=0; i<nums.size();++i){
            int s = target-nums[i];
            if (map.find(s)!=map.end()){
                return {map[s], i};
            }
            map.insert({nums[i], i});
        }
        return {};
    }
};